File: design-patterns/go/adapter/transport/helicopter.go
// helicopter.go

package main

type Helicopter struct {
}

func NewHelicopter() (helicopter *Helicopter) {
	helicopter = &Helicopter{}
	return
}


func (helicopter *Helicopter) GetNumberOfEngines() int {
	return 1
}

func (helicopter *Helicopter) GetNumberOfWheels() int {
	return 0
}

func (helicopter *Helicopter) GetWeight() float64 {
	return 12_000
}

func (helicopter *Helicopter) GetDistanceTravelled() float64 {
	return 180
}


func (helicopter *Helicopter) GetTravelCostTotal() float64 {
	return 20_000
}

File: design-patterns/ts/adapter/transport/helicopter.ts
// helicopter.ts

import AirTransport from "./air-transport";

class Helicopter implements AirTransport {
    getNumberOfWheels(): number {
        return 0;
    }

    getNumberOfEngines(): number {
        return 1;
    }

    getWeight(): number {
        return 12000;
    }

    getDistanceTravelled(): number {
        return 180;
    }
    
    getTravelCostTotal(): number {
        return 20000;
    }
}

export default Helicopter;